static/js/composebox_typeahead.js

```diff
@@ -212,6 +217,15 @@ exports.tokenize_compose_str = function (s) {
     while (i > min_i) {
         i -= 1;
         switch (s[i]) {
+            case '`':
+            case '~':
+                // Code block must start on a new line
+                if (i === 2) {
+                    return s.slice(0);
+                } else if (s[i-3] === "\n") {
+                    return s.slice(i-2);
```

Comment:
Can you explain why this logic is correct?  It seems like `i-3` might underflow the array....